app/controllers/application_controller.rb

```diff
@@ -12,6 +12,12 @@ def authorize_request
     render json: { message: 'Unauthorized' }, status: :unauthorized
   end
 
+  def authorize_admin
+    return if current_user.role == 'admin'
```

Comment:
Acho que eu colocaria um
```suggestion
    return if current_user.role.to_s == 'admin'
```
só pra garantir 😄

Reply:
Ou melhor...

Move essa lógica pra dentro do model de User:
```rb
class User
  ADMIN_ROLE_NAME = 'admin'.freeze

  # ...

  def admin?
    role.to_s == ADMIN_ROLE
  end
end
```

E dai aqui vc faz:
```suggestion
    return if current_user.admin?
```

Reply:
Inclusive uma outra coisa que daria pra vc fazer... Que não é necessário dado o contexto da sua app, mas só pra vc saber...

Olha esse código aqui:
https://github.com/Leoruwer/expenses-manager-api/blob/aab3c0ca1fa00957e9cd1789b0e8bf488f36408c/app/controllers/authentication_controller.rb#L10

Vc poderia trocar ele para isso aqui:
```rb
token = JsonWebToken.encode(user_id: user.id, admin: user.admin?)
```

E aí vc sabe que o user é admin ou não pelo token... E não precisaria necessariamente verificar no banco de dados novamente, entendeu?

Daí aqui poderia ser tipo assim:
```suggestion
    return unless decoded_token[:admin]
```

Mais ou menos isso aí a idéia 😄

Reply:
Então, como eu to usando um `enum` nesse caso do User, o Rails faz sua mágica e já cria os metodos: `.admin?` e `.user?`
Então só adicionar uma validação de `current_user.admin?` já funciona